Build failure: libfm

## Steps To Reproduce

Steps to reproduce the behavior:

```console
$ nix build github:nixos/nixpkgs#libfm
```

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```console
$ nix build github:nixos/nixpkgs#libfm
error: builder for '/nix/store/jh1i9yz23sa8wgyamdcvd8drkf7g4rbz-libfm-1.3.2.drv' failed with exit code 2;
       last 25 log lines:
       >       |               ^~~~~~~~~~~~~~~~~~~~~~
       > action.c:933:42: note: expected 'gpointer' {aka 'void *'} but argument is of type 'gconstpointer' {aka 'const void *'}
       >   933 | static gpointer _g_object_ref0 (gpointer self) {
       >       |                                 ~~~~~~~~~^~~~
       > action.c: In function 'fm_load_all_actions':
       > action.c:2050:25: error: assignment to 'gchar **' {aka 'char **'} from incompatible pointer type 'const gchar * const*' {aka 'const char * const*'} [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types8;;]
       >  2050 |         _tmp2_ = _tmp1_ = g_get_system_data_dirs ();
       >       |                         ^
       > condition.c: In function 'fm_file_action_condition_match_folder':
       > condition.c:1348:33: warning: 'g_pattern_match_string' is deprecated: Use 'g_pattern_spec_match_string' instead [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations8;;]
       >  1348 |                                 _tmp16_ = g_pattern_match_string (_tmp14_, _tmp15_);
       >       |                                 ^~~~~~~
       > /nix/store/y8fxzs8srzd6d74a85zbpssjr8wkj9q4-glib-2.82.1-dev/include/glib-2.0/glib/gpattern.h:57:15: note: declared here
       >    57 | gboolean      g_pattern_match_string   (GPatternSpec *pspec,
       >       |               ^~~~~~~~~~~~~~~~~~~~~~
       > make[4]: *** [Makefile:554: libfmactions_la-action.lo] Error 1
       > make[4]: *** Waiting for unfinished jobs....
       > make[4]: Leaving directory '/build/libfm-1.3.2/src/actions'
       > make[3]: *** [Makefile:2901: all-recursive] Error 1
       > make[3]: Leaving directory '/build/libfm-1.3.2/src'
       > make[2]: *** [Makefile:1261: all] Error 2
       > make[2]: Leaving directory '/build/libfm-1.3.2/src'
       > make[1]: *** [Makefile:569: all-recursive] Error 1
       > make[1]: Leaving directory '/build/libfm-1.3.2'
       > make: *** [Makefile:471: all] Error 2
       For full logs, run 'nix log /nix/store/jh1i9yz23sa8wgyamdcvd8drkf7g4rbz-libfm-1.3.2.drv'.
```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

```console
$ nix why-depends --derivation .#nixosConfigurations.framework.config.system.build.toplevel .#nixosConfigurations.framework.pkgs.libfm
/nix/store/wa33ldjyf89ayra0g3985nfxyqzb1jyw-nixos-system-framework-25.05.20241227.7cc0bff.drv
└───/nix/store/qv2vx19sg72bk5psrs35wfds4vyvkc71-system-path.drv
    └───/nix/store/nncblm8mb4d51havrvzzjczdsqmvib6h-xdg-desktop-portal-lxqt-1.1.0.drv
        └───/nix/store/skl4hc6cva1hmy87s0s6g0h75yhqgisl-libfm-qt-2.1.0.drv
            └───/nix/store/jh1i9yz23sa8wgyamdcvd8drkf7g4rbz-libfm-1.3.2.drv
```

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.66, NixOS, 25.05 (Warbler), 25.05.20241219.d70bd19`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - nixpkgs: `/nix/store/8fwsiv0hd7nw1brkvka0jf1frk3m7qkr-source`

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@ttuegel

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/li/libfm/0001-fm-load-all-actions.patch
+++ b/pkgs/by-name/li/libfm/0001-fm-load-all-actions.patch
@@ -1,0 +1,11 @@
+--- a/src/actions/action.c
++++ b/src/actions/action.c
+@@ -2047,7 +2047,7 @@
+ 	gchar* _tmp9_ = NULL;
+ 	_tmp0_ = fm_all_actions;
+ 	g_hash_table_remove_all (_tmp0_);
+-	_tmp2_ = _tmp1_ = g_get_system_data_dirs ();
++	_tmp2_ = _tmp1_ = (gchar **)g_get_system_data_dirs ();
+ 	dirs = _tmp2_;
+ 	dirs_length1 = _vala_array_length (_tmp1_);
+ 	_dirs_size_ = dirs_length1;

--- a/pkgs/by-name/li/libfm/0002-exo-icon-view-key-press-event.patch
+++ b/pkgs/by-name/li/libfm/0002-exo-icon-view-key-press-event.patch
@@ -1,0 +1,11 @@
+--- a/src/gtk/exo/exo-icon-view.c
++++ b/src/gtk/exo/exo-icon-view.c
+@@ -2819,7 +2819,7 @@
+   /* allocate a new event to forward */
+   new_event = gdk_event_copy ((GdkEvent *) event);
+   g_object_unref (G_OBJECT (new_event->key.window));
+-  new_event->key.window = g_object_ref (G_OBJECT (gtk_widget_get_window (icon_view->priv->search_entry)));
++  new_event->key.window = (GdkWindow *)g_object_ref (G_OBJECT (gtk_widget_get_window (icon_view->priv->search_entry)));
+ 
+   /* send the event to the search entry. If the "preedit-changed" signal is
+    * emitted during this event, priv->search_imcontext_changed will be set.

--- a/pkgs/by-name/li/libfm/0003-ask-action-on-drop.patch
+++ b/pkgs/by-name/li/libfm/0003-ask-action-on-drop.patch
@@ -1,0 +1,11 @@
+--- a/src/gtk/fm-dnd-dest.c
++++ b/src/gtk/fm-dnd-dest.c
+@@ -458,7 +458,7 @@
+             gtk_action_set_sensitive(act, FALSE);
+         }
+     }
+-    ri.menu = g_object_ref(gtk_ui_manager_get_widget(ui, "/popup"));
++    ri.menu = (GtkMenu *)g_object_ref(gtk_ui_manager_get_widget(ui, "/popup"));
+     g_signal_connect(ri.menu, "selection-done", G_CALLBACK(gtk_widget_destroy), NULL);
+     unmap_handler = g_signal_connect(ri.menu, "unmap",
+                                      G_CALLBACK(run_unmap_handler), &ri);

--- a/pkgs/by-name/li/libfm/0004-create-icon-view.patch
+++ b/pkgs/by-name/li/libfm/0004-create-icon-view.patch
@@ -1,0 +1,11 @@
+--- a/src/gtk/fm-standard-view.c
++++ b/src/gtk/fm-standard-view.c
+@@ -637,7 +637,7 @@
+                                 "text", FM_FOLDER_MODEL_COL_NAME );
+     if(fv->renderer_text)
+         g_object_unref(fv->renderer_text);
+-    fv->renderer_text = g_object_ref_sink(render);
++    fv->renderer_text = (FmCellRendererText *)g_object_ref_sink(render);
+     exo_icon_view_set_search_column((ExoIconView*)fv->view, FM_FOLDER_MODEL_COL_NAME);
+     g_signal_connect(fv->view, "item-activated", G_CALLBACK(on_icon_view_item_activated), fv);
+     g_signal_connect(fv->view, "selection-changed", G_CALLBACK(on_sel_changed), fv);

--- a/pkgs/by-name/li/libfm/package.nix
+++ b/pkgs/by-name/li/libfm/package.nix
@@ -27,6 +27,13 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-pQQmMDBM+OXYz/nVZca9VG8ii0jJYBU+02ajTofK0eU=";
   };
 
+  patches = [
+    ./0001-fm-load-all-actions.patch
+    ./0002-exo-icon-view-key-press-event.patch
+    ./0003-ask-action-on-drop.patch
+    ./0004-create-icon-view.patch
+  ];
+
   nativeBuildInputs = [
     vala
     pkg-config

